afs/afs-storage-api/src/main/java/com/powsybl/afs/storage/AbstractAppStorage.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.storage;
+
+import com.powsybl.afs.storage.events.*;
+import com.powsybl.commons.util.WeakListenerList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+public abstract class AbstractAppStorage implements AppStorage, ListenableAppStorage {
```

Comment:
ListenableAppStorage has to be removed definitely. 
Listeners should be managed in EventStore and not in AppStorage.